### DISCUSSION TITLE
How to validate kratos JWT tokens with Nginx Ingress?

### DISCUSSION THREAD

**Discussion by tobiasz-gleba** [ory/kratos#1954] [Q&A] [answered]
Plz help,
My goal is to validate Kratos JWT Tokens on Nginx Ingress layer and forward user id in header for my microservice. 
How???
https://github.com/ory/oathkeeper/issues/263

**Comment by vinckr** [comment, marked as answer]
That is a good question @TobiaszG24 
Apologies for the late response. 

We don't have a great step-by-step guide for this (yet), but maybe this post from the community can help: 
https://gruchalski.com/posts/2021-05-20-figuring-out-oathkeeper/

**Reply by harnash** [reply]
One solution would be to use Lua/OpenResty to implement this logic. We did that before and it worked. It is not ideal since you need to maintain this code in-sync with the backend but it is relatively simple.
One question would be how would one integrate it into k8s ingress?

https://github.com/openresty/lua-nginx-module#readme

**Comment by vinckr** [comment]
Hello @tobiasz-gleba 
Not sure if you are still stuck on this, but we have some examples for oathkeeper also with nginx here: https://github.com/ory/examples/tree/master/oathkeeper
I will mark this as answered for now, but feel free to come back to the discussion any time.
